test/framework/cluster.go

```diff
@@ -90,6 +94,38 @@ func NewClusterE2ETest(t *testing.T, provider Provider, opts ...ClusterE2ETestOp
 	return e
 }
 
+func WithHardware(vendor string, requiredCount int) ClusterE2ETestOpt {
+	return func(e *ClusterE2ETest) {
+		csvFilePath := os.Getenv(tinkerbellInventoryCsvFilePathEnvVar)
```

Comment:
Can we deal with reading the environment separately?

Reply:
We can but then every test would need to read the envvar and pass the value to the WithHardware filler when creating the ClusterE2ETest object. In principle I agree it should be an input, but in practice all the env vars required by ClusterE2ETest just read them within the implementation when required. Maybe this is a good option for a larger clean up task?